src/views/AddItem.jsx

```diff
@@ -16,14 +16,57 @@ export function AddItem({ listToken }) {
 	const handleSubmit = async (e) => {
 		e.preventDefault();
 
+		// const isValid = (itemName) => {
+		// 	checkEmpty(itemName);
+		// 	checkDuplicates(itemName);
+		// }
+
+		// const checkEmpty = (itemName) => {
+		// 	itemName.trim().length === 0;
+		// 	setMessage("Error: Please enter your item's name.");
+
+		// }
+
+		// const checkDuplicates = (itemName, itemList) => {
+		// 	const itemNames = itemList.map((item) => item.name.toLowerCase().replace(/[^a-zA-Z0-9 ]/g, ''),);
+		// 	itemNames.includes(itemName.toLowerCase().replace(/[^a-zA-Z0-9 ]/g, ''));
+		// 	setMessage(`${itemName} is already on your list. Please add a different item.`,);
+		// }
+
+		// try {
+		// 	isValid();
+		// 	addItem(listToken, {
+		// 		itemName,
+		// 		daysUntilNextPurchase,
+		// 	});
+		// 	setMessage(`${itemName} was successfully saved to your shopping list!`);
+		// } catch (error) {
+		// 	setMessage('Error: This item not added to the database.');
+		// }
+
+		const itemNames = itemList.map((item) =>
+			item.name.toLowerCase().replace(/[^a-zA-Z0-9 ]/g, ''),
+		); // Item names without special characters for comparison
+
 		try {
-			await addItem(listToken, {
-				itemName,
-				daysUntilNextPurchase,
-			});
-			setMessage(`${itemName} successfully saved to the database`);
+			// if (itemName.replace(/\s/g, '').length === 0) {
+			if (itemName.trim().length === 0) {
+				setMessage("Error: Please enter your item's name.");
+			} else if (
+				itemNames.includes(itemName.toLowerCase().replace(/[^a-zA-Z0-9 ]/g, ''))
+			) {
+				setMessage(
+					`${itemName} is already on your list. Please add a different item.`,
+				);
+			} else {
+				addItem(listToken, {
```

Comment:
If you truly want to catch errors from the `addItem` function you should add `await` before addItem. Because right now you are calling `addItem` and not waiting for it to succeed or fail so it just moves on and displays the success message. 
One create to create an error and check this out is to go in `addItem` and replace the `listCollectionRef` with a a plain string, for example "test".

Reply:
Done!